### ISSUE TITLE
Autocomplete widget not recognizing value change

### ISSUE THREAD

**Issue by schaunwheeler** [bokeh/bokeh#2259]
When I include the AutocompleteInput widget in an app and click on an item from the completions list, it doesn't seem to register the input. When I check the model (Bokeh.Collections('AutocompleteInput').models[0]), the 'value' attribute remains unchanged.

**Comment by rmelotte** [comment]
I still have this issue (or at least a very similar one) with 1.1.0dev11.

See this screencast:
![autocompleteinput](https://user-images.githubusercontent.com/20061471/55015150-9ad46e00-4fec-11e9-86e6-1f06d3bdbb06.gif)
The input gets filled with the correct value, but the "new" value that's passed to the on_change handler is the old value that the user typed.

I guess that there's some kind of race condition between the user triggering the "change" (by clicking outside the input) and the menu changing 'input_el.value'.

I found that doing the following solves the issue (but there is a caveat, see below)
In "autocompleteinput.ts", change the following:

Replace this: 
```
  protected _menu_click(event: MouseEvent): void {
    if (event.target != event.currentTarget && event.target instanceof Element) {
      this.input_el.value = event.target.textContent || ""
      this.input_el.focus()
      this._hide_menu()
    }
  }

```
with this:
```
  protected _menu_click(event: MouseEvent): void {
    if (event.target != event.currentTarget && event.target instanceof Element) {
      this.model.value = event.target.textContent || ""
      this.input_el.focus()
      this._hide_menu()
    }
  }

```

Doing this change result in the correct behavior, but the change listener end up being triggered two times: one time with what the user typed, and one time with the completion value.

However we may not be able to avoid it, as the user triggers an update himself by clicking outside the input.
Do you have another idea on how to fix it ? (to be clear: I know nothing about TypeScript so there may be another obvious solution)

**Comment by bryevdv** [comment]
This was fixed in #8829
